### PR TITLE
Base64 support for fbench

### PR DESCRIPTION
**Pull request by ohad7**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

Added a -d option to fbench to allow base64 decode before sending the request body. This is useful for cases where the POST body is binary e.g protobuf and we want to avoid the base64 decode on the client

### CODE REVIEW

File: fbench/src/fbench/client.cpp [hunk 1]
```diff
@@ -220,8 +224,15 @@ Client::run()
                 strcat(_linebuf, _args->_queryStringToAppend.c_str());
             }
             int cLen = _args->_usePostMode ? urlSource.nextContent() : 0;
+            auto content = urlSource.content();
```
Comment (baldersheim) [inline fbench/src/fbench/client.cpp]: Specify type explicit as it is not obvious what it is. You are also reassigning it further down.

File: fbench/src/fbench/client.cpp [hunk 2]
```diff
@@ -220,8 +224,15 @@ Client::run()
...
+            if (_args->_usePostMode && _args->_base64Decode) {
+                auto base64_decoded = Base64::decode(std::string(content, cLen));
+                content = base64_decoded.c_str();
```
Comment (baldersheim) [inline fbench/src/fbench/client.cpp]: Here you have a lifetime issue. **base64_decoded** must live at least as long as **content**.

File: fbench/src/fbench/client.cpp [hunk 3]
```diff
@@ -220,8 +224,15 @@ Client::run()
                 strcat(_linebuf, _args->_queryStringToAppend.c_str());
             }
             int cLen = _args->_usePostMode ? urlSource.nextContent() : 0;
+            auto content = urlSource.content();
+            if (_args->_usePostMode && _args->_base64Decode) {
+                auto base64_decoded = Base64::decode(std::string(content, cLen));
```
Comment (baldersheim) [inline fbench/src/fbench/client.cpp]: You should not wrap the input as an std::string. That will make a copy of the content. Just use the the decode(const char *, int length) method.

**Review by baldersheim — CHANGES_REQUESTED** [review]

File: fbench/src/fbench/client.cpp (hunk 2 above)
Comment (ohad7) [reply fbench/src/fbench/client.cpp]: Java developer negligence... fixed your other comments as comments as well.

**Review by baldersheim — APPROVED** [review]

**Pull request merged**